sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/VerifyMigrationDriver.kt

```diff
@@ -0,0 +1,29 @@
+package app.cash.sqldelight.gradle
+
+import java.sql.Driver
+import java.sql.DriverManager
+import java.util.ServiceLoader
+
+/**
+ * Interface used to mark a custom subclass of [java.sql.Driver] to be loaded by the isolated
+ * ClassLoader of the [VerifyMigrationTask]'s work queue prior to running migration verifications.
+ *
+ * On the consumer side, ensure that:
+ *   1. The driver is registered with java's [ServiceLoader] mechanism by placing a file
+ *   app.cash.sqldelight.gradle.VerifyMigrationDriver into resources/META-INF.services folder of
+ *   your custom driver library. The file should contain a binary name of the concrete provider
+ *   class. See [ServiceLoader] documentation for more details.
+ *   2. Your driver gets registered with the [DriverManager] in the
+ *   static (java) or init (kotlin) blocks so that it gets picked up when running
+ *   [VerifyMigrationTask].
+ *
+ * For example,
+ *
+ *   class CustomVerifyMigrationDriver : VerifyMigrationDriver {
+ *     init {
+ *       // Optionally, unregister other drivers (JDBC driver self-registers itself)
+ *       DriverManager.registerDriver(this)
+ *     }
+ *   }
+ */
+interface VerifyMigrationDriver: Driver
```

Comment:
Alright, I understand now. Instead of using a service loader is it possible to just have this type be another parameter to the verify migration task, I'm imagining an ideal API where you do something like

```groovy
tasks.withType<VerifyMigrationTask>().configureEach {
  driverInitializer = {
    DriverManager.registerDriver(...)
  }
}
```

Reply:
also I'd wait to write tests until we get this API where we want it to save yourself the trouble

Reply:
Ah interesting, adding `Serializable` interface to the `DriverInitializer` seems to do the trick

Reply:
With this approach, I no longer need to pass `connectionProperties`, so I think I'll just remove it

Reply:
if its not an input, does it work? The service loader implementation also wouldn't properly invalidate if you changed your implementation so I'm not too worried about that as a requirement

Reply:
(as in not annotated with @get:input)

Reply:
Hmm, I'm getting this after removing the `@get: Input`:
```
  - In plugin 'app.cash.sqldelight.gradle.SqlDelightPlugin$Inject' type 'app.cash.sqldelight.gradle.VerifyMigrationTask' property 'driverInitializer' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/8.0.2/userguide/validation_problems.html#missing_annotation for more details about this problem.
```
So perhaps we do need it?

Reply:
option 2? Marking it as `@Internal`? And then exposing a function on the task to set it

Reply:
👍 I didn't have to introduce another function to set it, it seems to work just as well

Reply:
Yeah, @Input is a required input that is checked and fails if it is not set. @Internal is optional.